common/chains/client/rpc_client.go

```diff
@@ -0,0 +1,83 @@
+package client
+
+import (
+	"context"
+	"math/big"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+)
+
+type RPCClient[
+	CHAINID types.ID,
+	SEQ types.Sequence,
+	ADDR types.Hashable,
+	BLOCK any,
+	BLOCKHASH types.Hashable,
+	TX any,
+	TXHASH types.Hashable,
+	EVENT any,
+	EVENTOPS any, // event filter query options
+	TXRECEIPT any,
```

Comment:
Could you investigate using ChainReceipt here?
https://github.com/smartcontractkit/chainlink/blob/171034e5ddded80d21ee8d4859c70dba4339e37b/common/txmgr/types/tx_store.go#L116

Should be ok to move ChainReceipt out to common/types
cc @aalu1418

Reply:
Just Implemented this change. I had to change a few things around to enable it to work.
1. I created a new interface in common/types called [chain_receipt.go](https://github.com/smartcontractkit/chainlink/blob/d0e470b9c5b9d273357c77f78863f7ceedbdc80a/common/types/chain_receipt.go#L5-L14)
2. I had to use CallContext to call eth_getTransactionReceipt rather than the geth client-specific TransactionReceipt() function so that I could return the evmtypes.Receipt. ([rpc_client.go](https://github.com/smartcontractkit/chainlink/blob/d0e470b9c5b9d273357c77f78863f7ceedbdc80a/core/chains/evm/client/rpc_client.go#L322-L332))